blog-posts/boosting-developer-productivity-with-github-actions.md

```diff
@@ -0,0 +1,266 @@
+
+```ts
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import { lint, formatResult } from './lint';
+
+(async function run() {
+    const title = github.context.payload.pull_request?.title;
+    const configFile = core.getInput('commitlintConfigFile');
+
+    core.info(`🔎 Checking if the title of this PR "${title}" meets the requirements ...`);
+
+    try {
+        const lintResult = await lint(title, configFile);
+        if (!lintResult.valid) {
+            core.setFailed(`\n ${formatResult(lintResult)}`);
+        } else {
+            core.info('✔️ All good');
+        }
+    } catch (error) {
+        core.setFailed(error as Error);
+    }
+})();
+```
+
+We won't go into a lot of detail about what this code does. I'll leave this as homework for you to explore.
+
+The main takeaway should be, that there are two important libraries that we are using: `@actions/core` and `@actions/github`. These two libraries provide a lot of utilities to interact with the process that is running on the GitHub runner. They allow you to fetch information about the event that triggered this workflow as well as make it easy to report the result of your action. You can get up close and personal with this code and explore the inner workings in [the action's repository](https://github.com/satellytes/commitlint-pr-title).
+
+### Compiling and bundling an Action
+As we've seen in the previous subsection we can have pretty much any code we want, as long as we provide a main entry point for the GitHub runner to execute. There are thousands of ways to turn TypeScript into JavaScript. The one way that I found most comfortable was to use [`@vercel/ncc`](https://github.com/vercel/ncc) to transpile, compile and bundle this action. `ncc` is a CLI tool that supports TypeScript out of the box and is meant to ship a single file that contains everything that is needed, from your actual source code to everything from inside the depths of your `node_modules` folder that is required by your action.
+
+The TypeScript file shown above is called `src/main.ts`. The bundling is done by calling `ncc build src/main.ts`. This will produce a file `dist/index.js` that we then point to in the `action.yml` file, as shown above.
+
+There are a few pitfalls here:
+- If your action involves a build step (like compiling TypeScript to JavaScript), you need to remember to include the generated bundle in your repository, in this case the `dist` folder. For regular projects this is not the case most of the time. Generally we don't want to include generated code in our repository. In this case you need to resist your urge to add `dist` to your `.gitignore` file.
```

Comment:
Well... no. My assumption is, that every regular developer feels the urge to add `dist` to the `.gitignore` file. What I'm saying is that in the case of an action repository this should not be done. So the developer "needs to resists the urge to add `dist` to `.gitignore`". Maybe re-read with a clear focus... I claim it makes sense.

Reply:
Ohhh that way? Really? Because I got it upside down. I read:

> For regular projects this is not the case most of the time.

and I just nod, because you don't check in the dist folder in usual projects but you do that for libraries in general.

And then I just assumed that you talk about doing exactly that here. It's still confusing and maybe we have a chat on that before creating an endless thread 😄